### DISCUSSION TITLE
How to use ActionScript to determine if my app is debugging via VSCode + AS3MXML extension?

### DISCUSSION THREAD

**Discussion by ylazy** [BowlerHatLLC/vscode-as3mxml#757] [Q&A] [answered]
Hi!

I'm building a lib that need to know whether my app is debugging via VSCode and trace commands are output to the VSCode's Debug Console? I can't find a perfect way to detect this. Any ideas?

Thank you!

**Comment by joshtynjala** [comment, marked as answer]
This should tell you if you're running in the AIR Debug Launcher (ADL):

```as3
if (Capabilities.playerType == "Desktop" && Capabilities.isDebugger)
{
    // running in ADL
}
```

However, I don't think that there's a way to detect whether an IDE's debugger is actively connected to ADL or not.
